### PR TITLE
Junit testing

### PR DESCRIPTION
**Pull request by alexdavcudriz**
## Summary
- Controllers testing was created.
- Repositories testieng was created.
## Trello Card
https://trello.com/c/1NNpqbY9/28-junit-testing

### CODE REVIEW

File: src/test/java/wolox/training/controllers/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    void whenFindAll_thenReturnAllBooks() throws Exception {
+        when(bookService.findAll()).thenReturn(Collections.singleton(bookTest));
+        mockMvc.perform(get(BOOK_PATH)
+                .accept(MediaType.APPLICATION_JSON_VALUE)
+                .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andDo(print()).andExpect(status().isOk())
+                .andExpect(jsonPath("[0].genre").value(bookTest.getGenre()))
+                .andExpect(jsonPath("[0].author").value(bookTest.getAuthor()))
+                .andExpect(jsonPath("[0].title").value(bookTest.getTitle()))
+                .andExpect(jsonPath("[0].subtitle").value(bookTest.getSubtitle()));
```
Comment (facucompe) [inline src/test/java/wolox/training/controllers/BookControllerTest.java:71]:
```
.andExpect(content().json(objectMapper.writeValueAsString(bookTest)))
```

File: src/test/java/wolox/training/controllers/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    @Test
+    void whenDelete_thenReturnIsOk() throws Exception {
+        doNothing().when(bookService).delete(any());
+        mockMvc.perform(delete(BOOK_PATH + "/{id}", BOOK_ID)
+                .accept(MediaType.APPLICATION_JSON_VALUE)
+                .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andDo(print()).andExpect(status().isOk());
+    }
+
+}
```
Comment (facucompe) [inline src/test/java/wolox/training/controllers/BookControllerTest.java]:
```suggestion
}

```

File: src/test/java/wolox/training/controllers/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.models.Book;
+import wolox.training.services.BookService;
+
+@WebMvcTest(BookController.class)
+class BookControllerTest {
```
Comment (facucompe) [inline src/test/java/wolox/training/controllers/BookControllerTest.java:31]: What about not found cases

File: src/test/java/wolox/training/repositories/BookRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Test
+    void whenDelete_thenReturnZeroBooks() {
+        Book bookFound = bookRepository.findByAuthor(DEFAULT_AUTHOR).get();
+        bookRepository.delete(bookFound);
+        Iterable<Book> books = bookRepository.findAll();
+        assertThat(books.iterator().hasNext()).isFalse();
+    }
+
+}
```
Comment (facucompe) [inline src/test/java/wolox/training/repositories/BookRepositoryTest.java]:
```suggestion
}

```

**Review by facucompe — CHANGES_REQUESTED** [review]
Amazing work!!

**Review by facucompe — APPROVED** [review]

**Review by jdhiguita — APPROVED** [review]

**Pull request merged**